src/bindgen/cargo/cargo_metadata.rs

```diff
@@ -180,16 +180,51 @@ impl Eq for Dependency {}
 pub fn metadata(
     manifest_path: &Path,
     existing_metadata_file: Option<&Path>,
+    all_platforms: bool,
 ) -> Result<Metadata, Error> {
     let output;
     let metadata = match existing_metadata_file {
         Some(path) => Cow::Owned(std::fs::read_to_string(path)?),
         None => {
+            let mut target = None;
+
+            if !all_platforms {
+                // cargo metadata defaults to giving information for _all_ targets.
+                // We figure out the host platform through rustc and use that.
+                // We unfortunatelly cannot go through cargo, since cargo rustc _also_ builds.
+                // If `rustc` fails to run, we just fall back to not passing --filter-platforms.
+                //
+                // NOTE: We set the current directory in case of rustup shenanigans.
+                let rustc = Command::new("rustc")
+                    .current_dir(manifest_path.parent().unwrap())
+                    .arg("-vV")
+                    .output();
+                log::debug!("Discovering host platform by {:?}", rustc);
+                if let Ok(out) = rustc {
+                    if let Ok(stdout) = std::str::from_utf8(&out.stdout) {
+                        let field = "host: ";
```

Comment:
Hmm, this also seems pretty odd of a default... Thinking a bit more about this, shouldn't this default to fetch the _target_ dependencies?

Reply:
I'm not sure I follow? From what I can tell, cbindgen is never passed a target platform, so the only reasonable thing to use here is the host platform (which is what this code does). Or rather, it uses the `host` field of `rustc -vV` for whatever `rustc` cbindgen is told to use.

Reply:
Right, my point is that cbindgen doesn't have target information, but using the host also isn't the right thing if you run cbindgen as part of the build.

Reply:
Ah, you're imagining that someone runs, say, `cargo build --target xyz`, and then `build.rs` invokes `cbindgen`, and it ends up using the host target even though `xyz` was explicitly passed in?

It looks like we can use some of the environment variables [cargo sets for build scripts](https://doc.rust-lang.org/cargo/reference/environment-variables.html#environment-variables-cargo-sets-for-build-scripts) here. Specifically, the `CARGO_CFG_` variables. I'll give that a stab!

Reply:
Ah, or better yet, it also just sets `TARGET`.

Reply:
Take a look at https://github.com/eqrion/cbindgen/pull/676/commits/b5abdf6fe048116cf66f260e2c8627dfd0ef8b42 — I think that does what you want!